internal/cluster/distribution/eks/eksadapter/node_pool_manager.go

```diff
@@ -147,21 +149,29 @@ func (n nodePoolManager) ListNodePools(ctx context.Context, cluster cluster.Clus
 	describeStacksInput := cloudformation.DescribeStacksInput{}
 	nodePools := make([]eks.NodePool, 0, len(nodePoolNames))
 	for _, nodePoolName := range nodePoolNames {
+		nodePools = append(nodePools, eks.NodePool{
+			Name:   nodePoolName,
+			Labels: labelSets[nodePoolName],
+		})
+		nodePool := &nodePools[len(nodePools)-1]
+
 		stackName := generateNodePoolStackName(cluster.Name, nodePoolName)
 		describeStacksInput.StackName = &stackName
 		stackDescriptions, err := cfClient.DescribeStacks(&describeStacksInput)
 		if err != nil {
-			return nil, errors.WrapWithDetails(err, "retrieving node pool cloudformation stack failed",
-				"cluster", cluster,
-				"input", describeStacksInput,
-			)
+			nodePool.Status = eks.NodePoolStatusError
```

Comment:
It could be that as well.

This code is later modified to be more sophisticated, but your comment is still relevant in that context so I'm going to consider using unknown here instead of error, not decided yet.

Reply:
Yeah, I think unknown would fit better here.

Reply:
It could be that as well.

This code is later modified to be more sophisticated, but your comment is still relevant in that context so I'm going to consider using unknown here instead of error.

Yeah, I think unknown would fit better here.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/f513dae37072b38a31753d8ddb5a981e4c878eb9..c5c7b1f092402fc85cfe94368f1384ec9f5f5bb1).